src/Console/MoveToAvatarCache.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Console;
+
+use Friendica\App\BaseURL;
+use Friendica\Contact\Avatar;
+use Friendica\Core\L10n;
+use Friendica\Model\Contact;
+use Friendica\Model\Photo;
+use Friendica\Util\Images;
+use Friendica\Object\Image;
+
+/**
+ * tool to move cached avatars to the avatar file cache.
+ */
+class MoveToAvatarCache extends \Asika\SimpleConsole\Console
+{
+	protected $helpOptions = ['h', 'help', '?'];
+
+	/**
+	 * @var $dba Friendica\Database\Database
+	 */
+	private $dba;
+
+	/**
+	 * @var $baseurl Friendica\App\BaseURL
+	 */
+	private $baseurl;
+
+	/**
+	 * @var L10n
+	 */
+	private $l10n;
+
+	protected function getHelp()
+	{
+		$help = <<<HELP
+console movetoavatarcache - Move all cached avatars to the file based avatar cache
+Synopsis
+	bin/console movetoavatarcache
+
+Description
+	bin/console movetoavatarcache
+		Move all cached avatars to the file based avatar cache
+
+Options
+	-h|--help|-? Show help information
+HELP;
+		return $help;
+	}
+
+	public function __construct(\Friendica\Database\Database $dba, BaseURL $baseurl, L10n $l10n, array $argv = null)
+	{
+		parent::__construct($argv);
+
+		$this->dba     = $dba;
+		$this->baseurl = $baseurl;
+		$this->l10n    = $l10n;
+	}
+
+	protected function doExecute()
+	{
+		$condition = ["`avatar` != ? AND `photo` LIKE ? AND `uid` = ? AND `uri-id` != ? AND NOT `uri-id` IS NULL",
+			'', $this->baseurl->get() . '/photo/%', 0, 0];
+
+		$count    = 0;
+		$total    = $this->dba->count('contact', $condition);
+		$contacts = $this->dba->select('contact', ['id', 'avatar', 'photo', 'uri-id', 'url', 'avatar'], $condition, ['order' => ['id']]);
+		while ($contact = $this->dba->fetch($contacts)) {
+			echo ++$count . '/' . $total . "\t" . $contact['id'] . "\t" . $contact['url'] . "\t";
```

Comment:
Please use `$this->out()` for automatic line feed output.

Reply:
Problem is that I need unbuffered and immediate output without a line feed.

Reply:
Update: The function can be called without a line feed, so I was able changing it.